front/src/modules/trainschedule/components/TimetableV2/types.ts

```diff
@@ -0,0 +1,24 @@
+import type {
+  LightRollingStockWithLiveries,
+  SimulationSummaryResult,
+} from 'common/api/osrdEditoastApi';
+
+export type TrainScheduleWithDetails = {
+  id: number;
+  trainName: string;
+  startTime: string;
+  arrivalTime: string;
+  /**
+   * in ms
+   */
+  duration: number;
+  stepsCount: number;
+  pathLength: string;
+  rollingStock?: LightRollingStockWithLiveries;
+  mechanicalEnergyConsumed: number;
+  speedLimitTag: string | null;
+  labels: string[];
+  invalidReasons?: InvalidReason;
```

Comment:
I think we can only have 1 reason
```suggestion
  invalidReason?: InvalidReason;
```

Reply:
Yep you're right, we know have only one reason